src/groups.js

```diff
@@ -6,10 +6,21 @@ import * as cognito from "./cognito.js";
 import { NotFound } from './httpErrors.js';
 import { GroupSource } from "./sources/index.js";
 import { markUserStaleBeforeNow } from "./user.js";
+import { rootDirFullPath } from "./utils/index.js";
 
 import { readFile } from 'fs/promises';
 const GROUPS_DATA = JSON.parse(await readFile(GROUPS_DATA_FILE));
 
+/**
+ * Summary string of where the groups known to the server come from
+ * (useful for debugging purposes)
+ */
+export function groupsSummary() {
+  const nPublic = GROUPS_DATA.filter((g)=>g.isPublic).length;
+  const groupsFile = GROUPS_DATA_FILE.replace(rootDirFullPath, '.'); // shown relative to server.js
```

Comment:
Prefer path-handling functions for paths instead of string-handling functions.

```suggestion
  const groupsFile = path.relative(rootDirFullPath, GROUPS_DATA_FILE); // shown relative to server.js
```

Reply:
Thanks. Fixed in repush.